extensions/amp-video/0.1/video-cache.js

```diff
@@ -85,7 +88,27 @@ function selectVideoSource(videoEl) {
  */
 function applySourcesToVideo(videoEl, sources, maxBitrate) {
   sources
-    .sort((a, b) => a['bitrate_kbps'] - b['bitrate_kbps'])
+    .sort((a, b) => {
+      // 'codec' values can contain metadata after the '.' that we must strip
+      // for sorting purposes. For example, "vp09.00.30.08" contains level,
+      // profile, and color depth values that are ignored in this sort.
+      const aCodec = a['codec']?.split('.')[0];
+      const bCodec = b['codec']?.split('.')[0];
+
+      for (let i = 0; i < CODECS_IN_DESCENDING_PRIORITY.length; i++) {
```

Comment:
Not that it really matters given the super small array size, but have you considered doing something like
`return CODECS_IN_DESCENDING_PRIORITY.indexOf(bCodec) - CODECS_IN_DESCENDING_PRIORITY.indexOf(aCodec)`


which would give a block looking like

```
// Sort by bitrate
if (aCodec === bCodec)
  return aBitrate - bBitrate

// Sort by codec
return CODECS_IN_DESCENDING_PRIORITY.indexOf(bCodec) - CODECS_IN_DESCENDING_PRIORITY.indexOf(aCodec)
```

Reply:
I think using `indexOf()` is a good idea, thanks! I've updated the code to take this approach while also changing `CODECS_IN_DESCENDING_PRIORITY` to `CODECS_IN_ASCENDING_PRIORITY`, allowing the returned indices to properly reflect priority order (e.g., `-1` will represent the least preferred options). For readability purposes, I didn't do an early exit for the case where the two codec values are equal, but I'll add the early exit if you feel strongly about it.